hooks/pre_gen_project.py

```diff
@@ -17,4 +17,11 @@
     print(f"ERROR: '{package_name}' is not a valid Python module name.")
     sys.exit(1)
 
+# TODO temporary, until GitLab CI support for Semantic Release will be added
+ci_type = '{{ cookiecutter.ci }}'
+versioning_approach = '{{ cookiecutter.versioning }}'
+if ci_type == 'GitLab' and versioning_approach == "Python Semantic Release":
+    print(f"ERROR: 'Python Semantic Release' is currently not supported for projects with GitLab CI.")
+    sys.exit(1)
```

Comment:
I would suggest adding Python Semantic as the option for versioning approach only after adding methods for supporting it instead of printing some errors

Reply:
Good idea!

Reply:
No longer applies after _Python Semantic Release_ got scraped from this PR.